ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/VirtualArtifactRepository.java

```diff
@@ -123,8 +127,17 @@ private void transferArtifact(Object artifactModel, OutputStream destination) th
 			if (location == null) {
 				throw new FileNotFoundException(bundleInfo.getSymbolicName());
 			}
-			try (InputStream is = location.toURL().openStream()) {
-				is.transferTo(destination);
+
+			File bundleLocation = new File(location);
+			if (bundleLocation.isFile()) {
+				try (InputStream is = new FileInputStream(bundleLocation)) {
+					is.transferTo(destination);
+				}
+			} else {
+				ZipOutputStream zip = new ZipOutputStream(destination);
```

Comment:
A few lines downwards we have used a `JarOutputStream`.
Is there a reason for the difference? I wonder if just a ZipOutputStream would be sufficient down there?
Looking at `JarOutputStream` the main addition to `ZipOutputStream` seems to be that it adds a `META-INF/MANIFEST.MF` entry and has a special magic-number for zips.